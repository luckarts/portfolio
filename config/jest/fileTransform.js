const path = require('path');

// This is a custom Jest transformer turning file imports into filenames.
// https://jest-bot.github.io/jest/docs/webpack.html

module.exports = {
  process(src, filename) {
    return `module.exports = ${JSON.stringify(path.basename(filename))};`;
  }
};