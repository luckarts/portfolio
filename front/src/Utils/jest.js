import checkPropTypes from "check-prop-types";
import { createStore } from "redux";
import rootReducer from "store/reducer";

export const checkProps = (component, expectedProps) => {
  // @ts-ignore
  const ErrProps = checkPropTypes(
    component.propTypes,
    expectedProps,
    "props",
    component.name
  );

  return ErrProps;
};
export const testStore = initialState => {
  return createStore(rootReducer, initialState);
};
