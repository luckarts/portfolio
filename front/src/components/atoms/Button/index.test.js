import React from "react";
import { shallow, mount } from "enzyme";
import { Button } from ".";
import { checkProps } from "Utils/jest";
let wrapper, props;
props = {
  children: "test"
};

it("Should render.", () => {
  expect(mount(<Button {...props} />)).toMatchSnapshot();
});

describe("Galerie Component with children props", () => {
  beforeEach(() => {
    wrapper = shallow(<Button {...props} />);
  });

  it("renders children when passed in", () => {
    expect(wrapper.find(".btn").text()).toEqual(props.children);
  });
  it(" Checking PropTypes no error", () => {
    const PropsError = checkProps(Button, props);
    expect(PropsError).toBeUndefined();
  });
});

describe("Galerie Component with onClick() props", () => {
  props = {
    children: "test",
    onClick: jest.fn()
  };
  it("Should render.", () => {
    expect(mount(<Button {...props} />)).toMatchSnapshot();
  });
  beforeEach(() => {
    wrapper = shallow(<Button {...props} />);
    wrapper.find(".btn").simulate("click");
  });
  it(" Checking PropTypes no error", () => {
    const PropsError = checkProps(Button, props);
    expect(PropsError).toBeUndefined();
  });

  it("call function onClick callback", () => {
    expect(props.onClick).toHaveBeenCalled();
  });
});
