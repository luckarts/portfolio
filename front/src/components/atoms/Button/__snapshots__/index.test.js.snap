// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Galerie Component with onClick() props Should render. 1`] = `ReactWrapper {}`;

exports[`Should render. 1`] = `ReactWrapper {}`;

exports[`renders correctly 1`] = `[Function]`;
